src/transformers/models/vibevoice/modular_vibevoice.py

```diff
@@ -0,0 +1,648 @@
+        Language modeling loss (for next-token prediction).
+    diffusion_loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `labels` for diffusion are provided):
+        Diffusion head loss (for acoustic token prediction).
+    logits (`torch.FloatTensor` of shape `(batch_size, sequence_length, config.vocab_size)`):
+        Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
+    past_key_values (`Cache`, *optional*, returned when `use_cache=True` is passed or when `config.use_cache=True`):
+        It is a [`~cache_utils.Cache`] instance. For more details, see our [kv cache guide](https://huggingface.co/docs/transformers/en/kv_cache).
+
+        Contains pre-computed hidden-states (key and values in the self-attention blocks) that can be used (see
+        `past_key_values` input) to speed up sequential decoding.
+    last_hidden_state (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`):
+        The hidden states at the last layer of the model.
+    attentions (`tuple(torch.FloatTensor)`, *optional*, returned when `output_attentions=True` is passed or when `config.output_attentions=True`):
+            Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size, num_heads, sequence_length,
+            sequence_length)`.
+
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+    """
+    loss: Optional[torch.FloatTensor] = None
+    diffusion_loss: Optional[torch.FloatTensor] = None
+    logits: Optional[torch.FloatTensor] = None
+    past_key_values: Optional[tuple[tuple[torch.FloatTensor]]] = None
+    last_hidden_state: Optional[torch.FloatTensor] = None
+    attentions: Optional[tuple[torch.FloatTensor, ...]] = None
+
+
+class VibeVoiceRMSNorm(Qwen2RMSNorm):
+    pass
+
+
+class TimestepEmbedder(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.layer_1 = nn.Linear(config.frequency_embedding_size, config.hidden_size, bias=False)
+        self.act = ACT2FN[config.hidden_act]
+        self.layer_2 = nn.Linear(config.hidden_size, config.hidden_size, bias=False)
+        self.frequency_embedding_size = config.frequency_embedding_size
+
+    def forward(self, timesteps):
+        # NOTE (ebezzam) using diffuser method like below instead of their custom method: https://github.com/pengzhiliang/transformers/blob/6e6e60fb95ca908feb0b039483adcc009809f579/src/transformers/models/vibevoice/modular_vibevoice_diffusion_head.py#L66
+        requires_backends(self, ["diffusers"])
+        t_freq = diffusers.models.embeddings.get_timestep_embedding(
+            timesteps=timesteps,
+            embedding_dim=self.frequency_embedding_size,
+            flip_sin_to_cos=True,
+            downscale_freq_shift=0,
+            scale=1.0,
+            max_period=10000,
+        ).to(timesteps.dtype)
```

Comment:
Using this method from the diffusers library to embed the timestep: https://github.com/huggingface/diffusers/blob/df8dd778177c3d2272f74cbebd880d7abd9f5ec9/src/diffusers/models/embeddings.py#L27